### DISCUSSION TITLE
OpenStack Swift segments are not removed when deleting a large object file (SLO)

### DISCUSSION THREAD

**Discussion by XierraRunnis** [iterate-ch/cyberduck#14465] [Q&A]
Hello group,
 
My English is not so good but I'll give it a try.

I use Cyberduck with OpenStack Swift on Informaniak Swiss Backup.

When I upload a larger file it is stored in 2 directories (default and default_segments). In default is the correct name with 0 byte and under default_segments you see the individual split parts in subdirectories that look like this for example: 2242a4bd7d293a3930f7e38403c0b594
 
Uplaod and download work without any problems.
 
Now to my problem. If I delete the file under default, the individual parts of the file remain under default_segments and are not deleted.  The storage space is not released. 
 
In order to delete a file completely, I first have to look for the folder under default_segements under Info -> Metadata, then delete it and then the actual file under default.
 
Unfortunately, this is very tedious and time-consuming, especially if you have a lot of files.

David Kocher wrote to me: When deleting the manifest files for a large upload we ought to collect all segment files and delete these as well.

Now i enclosed the LOG from the Cyberduck and some Screenshots. I use Cyberduck on Windows 10 21H2 the Cyberduck-Version is 8.3.2.37449.

I have carried out the following:

1. display the file 03_LW_E63.adi incl. info because of the directory for the segments. (Cyberduck01.png)

2. after deleting the file 03_LW_E63.adi, it is gone in default. however, the associated directory in default_segments is still present and also occupies the space. (essafb41...) (Cyberduck_02.png).

3. delete the segments via delete essafb4... (Cyberduck_03.png)

4. after manual deletion everything is gone. (Cyberduck_04.png)

Many greetings
BIGAIRFOX

This ticket system is to report bugs and feature requests. For support, visit the [help page](https://cyberduck.io/help/) first.

 * If you have trouble connecting to a server or your login credentials are not valid, try to resolve the issue with the assistance of your hosting service provider. Any such ticket will be closed with a resolution of _thirdparty_.

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. macOS]

**Log Files**
Please include the [log file](https://docs.cyberduck.io/cyberduck/support/#logging-output). You can reach the logging output in _Preferences → Connection_.

**Additional context**
[cyberduck.zip](https://github.com/iterate-ch/cyberduck/files/8462084/cyberduck.zip)
![Cyberduck_01](https://user-images.githubusercontent.com/103412466/162690618-a082bf1b-0b9a-4f4f-b90a-d3e41d2e0fe0.png)
![Cyberduck_02](https://user-images.githubusercontent.com/103412466/162690625-ab22ac83-2fa9-4ff7-a8bb-11fa6c1c5dae.png)
![Cyberduck_03](https://user-images.githubusercontent.com/103412466/162690634-caa87dd5-b6d9-4b2e-8b81-443e65616704.png)
![Cyberduck_04](https://user-images.githubusercontent.com/103412466/162690641-dcabc04b-af3d-401c-8f93-147be7cd28d0.png)

**Comment by dkocher** [comment]
Handling the deletion of large segments when replacing a file has been revised with #14327.

**Comment by dkocher** [comment]
Large object segments are stored in `.file-segments/` by default. You may want to reach out to Infomaniak support about `default_segments`.
